core/shared/src/main/scala/fs2/text.scala

```diff
@@ -212,60 +212,19 @@ object text {
       (out, carry)
     }
 
-    def extractLines(
-        buffer: Vector[String],
-        chunk: Chunk[String],
-        pendingLineFeed: Boolean
-    ): (Chunk[String], Vector[String], Boolean) = {
-      @tailrec
-      def go(
-          remainingInput: Vector[String],
-          buffer: Vector[String],
-          output: Vector[String],
-          pendingLineFeed: Boolean
```

Comment:
Does the new implementation handle the case where a `\r\n` sequence is split across chunks, with `\r` as final char in previous chunk and `\n` as first char in new chunk? In particular, this needs to be handled as a single newline.

Reply:
I think this test handles the case
```
property("grouped in 3 characater chunks") {
      forAll { (lines0: Stream[Pure, String]) =>
        val lines = lines0.map(escapeCrLf)
        val s = lines.intersperse("\r\n").toList.mkString.grouped(3).toList
        if (s.isEmpty)
          assertEquals(Stream.emits(s).through(text.lines).toList, Nil)
        else {
          assertEquals(Stream.emits(s).through(text.lines).toList, lines.toList)
          assertEquals(Stream.emits(s).unchunk.through(text.lines).toList, lines.toList)
        }
      }
    }
```

Reply:
```
package fs2

object TextExplicitTest {

  def check() = {
    val stream = Stream.chunk(Chunk.singleton("abc\r")) ++ Stream.chunk(Chunk.singleton("\ndef"))
    val result = stream.through(text.lines[Pure]).compile.toList
    assert(List("abc", "def") == result)
    println(s"result: $result")
  }
}
```

```
> coreJVM/ test:console

scala> fs2.TextExplicitTest.check()
result: List(abc, def)
```